### ISSUE TITLE
Deprecate twisted.web.google

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5209]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5209|
|Type|task|
|Created|2011-07-28 21:44:21Z|
|Branch|https://github.com/twisted/twisted/tree/deprecate-google-5209|

The `twisted.web.google` module is a meager layer on top of the old HTTP client.  It's untested, undocumented, and not particularly useful.  Google also offers actual APIs for doing what it does.

The whole module should be deprecated.

Attachments:

* [5209.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/677/677f83d34efb2851a91a63f87ce6e0f842f97166/e2462875fbaf415479349a842918a91e7351a5c4.patch) (1829 bytes) - added by retenodus on 2011-07-29 01:01:29Z - Module deprecated
* [5209-2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/677/677f83d34efb2851a91a63f87ce6e0f842f97166/4509fcb6619dae30421ac276b2cb5e1386972942.patch) (1628 bytes) - added by retenodus on 2011-08-10 01:19:18Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5209 5209
type__task task
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_deprecate_google_5209 branches/deprecate-google-5209
branch_author__retenodus retenodus
status__closed closed
resolution__fixed fixed
component__web web
keywords__easy easy
time__1311889461000000 1311889461000000
changetime__1313271159000000 1313271159000000
version__None None
owner__exarkun exarkun
cc__jknight cc__retenodus cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to retenodus|
|-|-|

Thanks!

  1. Please generate patches from the top of the source tree.
  1. I think the entire `twisted.web.google` module should have the deprecation, not the two names defined in it.  See `twisted/enterprise/util.py` for an example.   For testing, the import can be nested inside the test method to trigger the deprecation (and an import-time deprecation is just what we want users to see, so this makes sense).  You may need to delete the module from `sys.modules` first to ensure it really gets imported (and to make the test work with `trial --until-failure`).
  1. Function arguments should have a space following the comma, eg `Version('Twisted', 11, 1, 0)`
  1. Test methods should always have a docstring explaining what is being tested.

Thanks again!

**Comment by twisted-trac** [comment]
|<img alt="retenodus's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|retenodus commented|
|-|-|

The new patch should be OK

**Comment by twisted-trac** [comment]
|<img alt="retenodus's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|retenodus removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set owner to retenodus|
|-|-|

The [doc/web/examples/google.py example](doc/web/examples/google.py example) also needs to go if it's deprecated imo.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

I don't particularly care in this case, but in general we do not remove documentation when we deprecate something.  We remove documentation when we _remove_ the thing.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

(In [32446]) Branching to 'deprecate-google-5209'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

(In [32447]) Apply most of 5209-2.patch

refs [#5209](https://github.com/twisted/twisted/issues/5209)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set owner to @exarkun|
|-|-|

I had some trouble applying this patch.  Visual inspection doesn't reveal any obvious issues, but the patch program refused to apply the `test_web.py` and `google.py` hunks.  Since it was a short patch, I applied it by hand.  I suggest trying to figure out why it can't be applied automatically though, since this could be a bigger problem for bigger patches.

Some comments about the contents of the patch:

  1. The `google.py` changes look superfluous now, since the deprecation is done entirely in `__init__.py`.  So I'll just drop them.
  1. A couple new lines have trailing whitespace, which we try to avoid, so I'll strip it from them.
  1. Deprecations should get a `.removal` news fragment (documented at ReviewProcess#Newsfiles).

Since these are simple, I'll make the adjustments and then merge.  Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun commented|
|-|-|

Jesus H, what a failure.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun set status to `closed`|
|-|-|

(In [32449]) Merge deprecate-google-5209

Author: retenodus
Reviewer: thijs, exarkun
Fixes: [#5209](https://github.com/twisted/twisted/issues/5209)

Deprecate all of `twisted.web.google`.
